tests/ExpressionParserTest.php

```diff
@@ -411,6 +415,51 @@ public function testUnknownTestWithoutSuggestions()
         $parser->parse($env->tokenize(new Source('{{ 1 is foobar }}', 'index')));
     }
 
+    public function testCompiledCodeForDynamicTest()
+    {
+        $env = new Environment(new ArrayLoader(['index' => '{{ "a" is foo_foo_bar_bar }}']), ['cache' => false, 'autoescape' => false]);
+        $env->addExtension(new class() extends AbstractExtension {
+            public function getTests()
+            {
+                return [
+                    new TwigTest('*_foo_*_bar', function ($foo, $bar, $a) {}),
+                ];
+            }
+        });
+
+        $this->assertStringContainsString('$this->env->getTest(\'*_foo_*_bar\')->getCallable()("foo", "bar", "a")', $env->compile($env->parse($env->tokenize(new Source($env->getLoader()->getSourceContext('index')->getCode(), 'index')))));
+    }
+
+    public function testCompiledCodeForDynamicFunction()
+    {
+        $env = new Environment(new ArrayLoader(['index' => '{{ foo_foo_bar_bar("a") }}']), ['cache' => false, 'autoescape' => false]);
+        $env->addExtension(new class() extends AbstractExtension {
+            public function getFunctions()
+            {
+                return [
+                    new TwigFunction('*_foo_*_bar', function ($foo, $bar, $a) {}),
+                ];
+            }
+        });
+
+        $this->assertStringContainsString('$this->env->getFunction(\'*_foo_*_bar\')->getCallable()("foo", "bar", "a")', $env->compile($env->parse($env->tokenize(new Source($env->getLoader()->getSourceContext('index')->getCode(), 'index')))));
+    }
+
+    public function testCompiledCodeForDynamicFilter()
+    {
+        $env = new Environment(new ArrayLoader(['index' => '{{ "a"|foo_foo_bar_bar }}']), ['cache' => false, 'autoescape' => false]);
+        $env->addExtension(new class() extends AbstractExtension {
+            public function getFilters()
+            {
+                return [
+                    new TwigFilter('*_foo_*_bar', function ($foo, $bar, $a) {}),
+                ];
+            }
+        });
+
+        $this->assertStringContainsString('$this->env->getFilter(\'*_foo_*_bar\')->getCallable()("foo", "bar", "a")', $env->compile($env->parse($env->tokenize(new Source($env->getLoader()->getSourceContext('index')->getCode(), 'index')))));
```

Comment:
Having the dynamic name here (at runtime) avoids matching the original `foo_foo_bar_bar` filter in ExtensionSet (not needed anyway as we just need the callable here).